docs/docs/creating-dynamically-rendered-navigation.md

```diff
@@ -2,7 +2,175 @@
 ---
 
-This is a stub. Help our community expand it.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Creating dynamic navigation in Gatsby
+
+At times you will want to be able to edit your website's navigation in response to a *change in requirements*. To achieve this, you can use Gatsby to dynamically generate your navigation. Where you store the data for your navigation can be anywhere - a backend API, CMS, headless CMS or even the filesystem.
+
+What this doc will cover:
+
+ - Adding data to your siteMetadata
+ - Querying it using GraphQL
+ - Pulling the data into a component using StaticQuery
+ - Dynamically generating a navigation component
+
+You will be using `gatsby-config.js` to store the data for your links. `gatsby-config.js` is a file used for configuring gatsby located in the root path of every Gatsby project. A plain old javascript object is exported from this file, this object contains the `siteMetadata` object which you can query through graphql when generating your static pages. 
+
+This guide will use the Gatsby starter project `gatsby-starter-default`, which can be downloaded through the Gatsby command line interface tool using the command `gatsby new [project-name] https://github.com/gatsbyjs/gatsby-starter-default#v2`.  
+
+### Creating the link data
+First, locate the `gatsby-config.js` file in the root directory of your project. Inside the `siteMetadata` object, add an array of menu link objects. These objects should contain two properties: name and link. Name is the name of your navigation item, link is the page which will be navigated to when a menu item is clicked.
+
+```js
+module.exports = {
+  siteMetadata: {
+    title: 'Gatsby Default Starter',
+    menuLinks:[
+      {name:'home',link:'/'},
+      {name:'page2',link:'/page-2'}
+    ]
+  },
+  plugins: []
+}
+```
+### Viewing the `siteMetadata` in GraphQL
+
+GraphQL can be used to query for information contained in the `siteMetadata` object located in your project's `gatsby-config.js`. In order to test this out you can start the `default-starter-project` in development mode by running `npm run develop`.
+
+Navigate to `http://localhost:8000/___graphql` in your browser to view the GraphiQL editor, which enables you to test GraphQL queries on the underlying APIs. Using the documentation explorer you can view the current GraphQL schema for your project which is an invaluable resource during development.
+
+Examining the available types in GraphQL you will notice that you can query `site`. This GraphQL type further returns the `siteMetadata` which needs to be accessed to create the dynamic navigation. At this point it is useful if you know a little GraphQL in order to extract the menu links. If you are unfamiliar with GraphQL, there is some excellent documentation available at (LINK) so you can brush up on your skills! The query below will return the menu links.
```

Comment:
Did you mean to add a link to this paragraph under (LINK)?

Reply:
Yes! I was just going to double check whether it should link internally to some documentation on Gatsby about GraphQL or if I should make it an external link to the official GraphQL website.

Reply:
I have added a link to https://graphql.org/learn/ let me know if this is allowed otherwise I can change it 👍